service/matching/taskListManager.go

```diff
@@ -323,6 +405,11 @@ func (c *taskListManagerImpl) getTask(ctx context.Context) (*getTaskResult, erro
 		}()
 	}
 
+	if ok, _ := c.rateLimiter.TryConsume(1); !ok {
+		// Note: Is a better error counter more appropriate here? We might not want high sevs for this
+		c.metricsClient.IncCounter(scope, metrics.PollErrorsCounter)
+		return nil, createServiceBusyError("TaskList dispatch exceeded limit")
```

Comment:
We want to block consumers on long poll and return empty polls to maintain the configured rate. We don't want them receive any busy errors.

Reply:
Good point. Fixed.